### PR TITLE
Frontend: Login

### PR DESCRIPTION
**Pull request by dylan-muszel**
### Summary
Login with front-end validations and navigation. 

### Screenshot
image

image

image

### Trello Card
https://trello.com/c/3OJkRJVV/19-u1-frontend-login-validaciones
https://trello.com/c/nbKvADQP/20-u1-frontend-login-navegaci%C3%B3n

### CODE REVIEW

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java [hunk 1]
```diff
@@ -0,0 +1,16 @@
+package ar.com.wolox.android.training.ui.errors;
+
+public class ErrorHandler {
+    public static String getErrorMessage(ErrorCode errorCode) {
+        switch (errorCode) {
+            case EMPTY_FIELDS:
+                return "Todos los campos son obligatorios";
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:7]: Try to get the text from the resources.

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+            case INVALID_EMAIL:
+                return "El email ingresado es incorrecto";
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:10]: Same here

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+        startActivity(intent);
+        getActivity().finish();
+    }
+
+    @OnClick(R.id.fragment_login_login)
+    public void onLogin() {
+        String email = mEmail.getText().toString();
+        String password = mPassword.getText().toString();
+
+        Boolean validateErrors =    this.validateEmptyFields(email, password)
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java]: Maybe its better to check empty email and if its a valid one , and then check if the password is empty. That allows the user to complete one first.

File: app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    @OnClick(R.id.fragment_login_signup)
+    public void onClickSignup() {
+        Intent intent = new Intent(getActivity(), SignupActivity.class);
+        startActivity(intent);
+    }
+
+    @OnClick(R.id.fragment_login_terms_conditions)
+    public void onClickTermsAndConditions() {
+        Intent intent = new Intent(Intent.ACTION_VIEW);
+        intent.setData(Uri.parse("https://www.wolox.com.ar/"));
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/login/LoginFragment.java]: Move this to resources or a constant . I prefer the first one , what do you think?

File: app/src/main/java/ar/com/wolox/android/training/ui/viewpager/random/RandomPresenter.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+
+import ar.com.wolox.android.training.model.ExampleModel;
+import ar.com.wolox.android.training.utils.UserSession;
+import ar.com.wolox.wolmo.core.presenter.BasePresenter;
+
+import java.util.Random;
+
+import javax.inject.Inject;
+
+public class RandomPresenter extends BasePresenter<IRandomView> {
```
Comment (DemianIac) [inline app/src/main/java/ar/com/wolox/android/training/ui/viewpager/random/RandomPresenter.java:13]: What is this? . I think that you don't use it anymore

**Review by DemianIac — CHANGES_REQUESTED** [review]
Some minor changes

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java (hunk 1 above)
Comment (dylan-muszel) [reply app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:7]: Its done later (https://github.com/wolox-training/dm-android/blob/home-page-3/app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java)

File: app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java (hunk 2 above)
Comment (dylan-muszel) [reply app/src/main/java/ar/com/wolox/android/training/ui/errors/ErrorHandler.java:10]: Same

File: app/src/main/java/ar/com/wolox/android/training/ui/viewpager/random/RandomPresenter.java (hunk 5 above)
Comment (dylan-muszel) [reply app/src/main/java/ar/com/wolox/android/training/ui/viewpager/random/RandomPresenter.java:13]: Its remove later

File: app/src/main/res/layout/custom_button_view.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content">
+    <Button
+        xmlns:android="http://schemas.android.com/apk/res/android"
+        android:id="@+id/custom_button"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:textColor="@android:color/white"
+        android:layout_marginTop="8dp"
```
Comment (DemianIac) [inline app/src/main/res/layout/custom_button_view.xml]: Remember not to use hardcode dimens . Create a new resource value in dimens.xml file and use it.

File: app/src/main/res/layout/fragment_login.xml [hunk 7]
```diff
@@ -1,59 +1,96 @@
...
+                android:layout_height="match_parent"
+                android:layout_width="match_parent"
+                android:alpha="0.7"
+                android:background="@drawable/fragment_login_gradient"/>
+        </FrameLayout>
+
+        <ImageView
+            android:layout_width="wrap_content"
+            android:layout_height="wrap_content"
+            android:layout_marginTop="100dp"
```
Comment (DemianIac) [inline app/src/main/res/layout/fragment_login.xml]: Same here of hardcode dimens

**Review by DemianIac — CHANGES_REQUESTED** [review]
Just need some minor changes. Please check the views and change the hardcode dimens , and use the resource file

**Review by DemianIac — APPROVED** [review]
gif

**Pull request merged**